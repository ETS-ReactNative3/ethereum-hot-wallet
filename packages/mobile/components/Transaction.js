/* global fetch */
// @flow
import * as React from 'react';
import { ScrollView } from 'react-native';
import List from 'antd-mobile-rn/lib/list/index.native';
import { API_DOMAIN } from 'dotenv.macro';

type Props = {
  transactionHash: string,
};

type State = {
  transaction: {
    blockHash: string,
    blockNumber: number,
    from: string,
    gas: number,
    gasPrice: string,
    hash: string,
    nonce: number,
    to: string,
    value: string,
  },
};

export default class Transaction extends React.Component<Props, State> {
  state = {
    transaction: {
      blockHash: '',
      blockNumber: 0,
      from: '',
      gas: 0,
      gasPrice: '',
      hash: '',
      nonce: 0,
      to: '',
      value: '',
    },
  };

  componentDidMount() {
    const { transactionHash } = this.props;
    fetch(`${API_DOMAIN}/transaction/${transactionHash}`)
      .then(response => response.json())
      .then(transaction => {
        this.setState({ transaction });
      });
  }

  props: Props;

  render() {
    const { transactionHash } = this.props;
    const { transaction } = this.state;

    return (
      <ScrollView
        automaticallyAdjustContentInsets={false}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
      >
        <List renderHeader={() => `Transaction ${transactionHash}`}>
          <List.Item extra={transaction.blockHash}>blockHash</List.Item>
          <List.Item extra={transaction.blockNumber}>blockNumber</List.Item>
          <List.Item extra={transaction.from}>from</List.Item>
          <List.Item extra={transaction.gas}>gas</List.Item>
          <List.Item extra={transaction.gasPrice}>gasPrice</List.Item>
          <List.Item extra={transaction.hash}>hash</List.Item>
          <List.Item extra={transaction.nonce}>nonce</List.Item>
          <List.Item extra={transaction.to}>to</List.Item>
          <List.Item extra={transaction.value}>value</List.Item>
        </List>
      </ScrollView>
    );
  }
}
