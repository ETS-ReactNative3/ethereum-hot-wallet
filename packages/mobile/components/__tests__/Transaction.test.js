// @flow
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Transaction from '../Transaction';

it('should render Transaction', async () => {
  const wrapper = shallow(<Transaction transactionHash="hash" />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
