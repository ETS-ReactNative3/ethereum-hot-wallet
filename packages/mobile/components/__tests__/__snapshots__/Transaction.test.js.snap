// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Transaction 1`] = `
<ScrollView
  automaticallyAdjustContentInsets={false}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
>
  <List
    renderHeader={[Function]}
    styles={
      Object {
        "Arrow": Object {
          "height": 13,
          "marginLeft": 8,
          "marginTop": 3,
          "width": 8,
        },
        "ArrowV": Object {
          "height": 8,
          "marginLeft": 8,
          "width": 13,
        },
        "Body": Object {
          "backgroundColor": "#ffffff",
          "borderTopColor": "#dddddd",
          "borderTopWidth": 0.5,
        },
        "BodyBottomLine": Object {
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#dddddd",
          "borderBottomWidth": 0.5,
          "bottom": 0,
          "height": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        "Brief": Object {
          "minHeight": 10,
        },
        "BriefText": Object {
          "color": "#888888",
          "fontSize": 15,
          "paddingTop": 3,
          "textAlignVertical": "center",
        },
        "Content": Object {
          "color": "#000000",
          "fontSize": 17,
          "textAlignVertical": "center",
        },
        "Extra": Object {
          "color": "#888888",
          "fontSize": 17,
          "textAlign": "right",
          "textAlignVertical": "center",
        },
        "Footer": Object {
          "backgroundColor": "#f5f5f9",
          "color": "#888888",
          "fontSize": 14,
          "paddingHorizontal": 15,
          "paddingVertical": 9,
        },
        "Header": Object {
          "backgroundColor": "#f5f5f9",
          "color": "#888888",
          "fontSize": 14,
          "paddingBottom": 9,
          "paddingHorizontal": 15,
          "paddingTop": 15,
        },
        "Item": Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flexDirection": "row",
          "flexGrow": 1,
          "paddingLeft": 15,
        },
        "Line": Object {
          "alignItems": "center",
          "borderBottomColor": "#dddddd",
          "borderBottomWidth": 0.5,
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 44,
          "paddingRight": 15,
          "paddingVertical": 6,
        },
        "Thumb": Object {
          "height": 22,
          "marginRight": 15,
          "width": 22,
        },
        "column": Object {
          "flex": 1,
          "flexDirection": "column",
        },
        "multipleLine": Object {
          "paddingVertical": 6,
        },
        "multipleThumb": Object {
          "height": 36,
          "width": 36,
        },
        "underlayColor": Object {
          "backgroundColor": "#dddddd",
        },
      }
    }
  >
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      blockHash
    </Item>
    <Item
      extra={0}
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      blockNumber
    </Item>
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      from
    </Item>
    <Item
      extra={0}
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      gas
    </Item>
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      gasPrice
    </Item>
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      hash
    </Item>
    <Item
      extra={0}
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      nonce
    </Item>
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      to
    </Item>
    <Item
      extra=""
      multipleLine={false}
      styles={
        Object {
          "Arrow": Object {
            "height": 13,
            "marginLeft": 8,
            "marginTop": 3,
            "width": 8,
          },
          "ArrowV": Object {
            "height": 8,
            "marginLeft": 8,
            "width": 13,
          },
          "Body": Object {
            "backgroundColor": "#ffffff",
            "borderTopColor": "#dddddd",
            "borderTopWidth": 0.5,
          },
          "BodyBottomLine": Object {
            "backgroundColor": "#ffffff",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          "Brief": Object {
            "minHeight": 10,
          },
          "BriefText": Object {
            "color": "#888888",
            "fontSize": 15,
            "paddingTop": 3,
            "textAlignVertical": "center",
          },
          "Content": Object {
            "color": "#000000",
            "fontSize": 17,
            "textAlignVertical": "center",
          },
          "Extra": Object {
            "color": "#888888",
            "fontSize": 17,
            "textAlign": "right",
            "textAlignVertical": "center",
          },
          "Footer": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingHorizontal": 15,
            "paddingVertical": 9,
          },
          "Header": Object {
            "backgroundColor": "#f5f5f9",
            "color": "#888888",
            "fontSize": 14,
            "paddingBottom": 9,
            "paddingHorizontal": 15,
            "paddingTop": 15,
          },
          "Item": Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "flexDirection": "row",
            "flexGrow": 1,
            "paddingLeft": 15,
          },
          "Line": Object {
            "alignItems": "center",
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "minHeight": 44,
            "paddingRight": 15,
            "paddingVertical": 6,
          },
          "Thumb": Object {
            "height": 22,
            "marginRight": 15,
            "width": 22,
          },
          "column": Object {
            "flex": 1,
            "flexDirection": "column",
          },
          "multipleLine": Object {
            "paddingVertical": 6,
          },
          "multipleThumb": Object {
            "height": 36,
            "width": 36,
          },
          "underlayColor": Object {
            "backgroundColor": "#dddddd",
          },
        }
      }
      wrap={false}
    >
      value
    </Item>
  </List>
</ScrollView>
`;
